src/query.rs

```diff
@@ -404,38 +403,43 @@ impl Query {
     // HD Wallets
     //
 
-    pub fn get_hd_wallets(&self) -> HashMap<Fingerprint, HDWallet> {
+    pub fn get_hd_wallets(&self) -> HashMap<DescriptorChecksum, HDWallet> {
         self.indexer.read().unwrap().watcher().wallets().clone()
     }
 
-    pub fn get_hd_wallet(&self, fingerprint: Fingerprint) -> Option<HDWallet> {
+    pub fn get_hd_wallet(&self, desc_check: DescriptorChecksum) -> Option<HDWallet> {
         self.indexer
             .read()
             .unwrap()
             .watcher()
-            .get(fingerprint)
+            .get(desc_check)
             .cloned()
     }
 
     // get the ScriptInfo entry of a derived hd key, without it necessarily being indexed
-    pub fn get_hd_script_info(&self, fingerprint: Fingerprint, index: u32) -> Option<ScriptInfo> {
+    pub fn get_hd_script_info(
+        &self,
+        desc_check: DescriptorChecksum,
+        index: u32,
+    ) -> Option<ScriptInfo> {
```

Comment:
Now that this depends on the RPC and can fail, the method should return a `Result<Option<ScriptInfo>>`